src/modules/search/components/OmniSearch.tsx

```diff
@@ -11,37 +11,82 @@ import {
   CircularProgress,
   List,
 } from '@mui/material';
-import { flatten, isEmpty } from 'lodash-es';
-import React, { useState } from 'react';
+import { flatten, isEmpty, sortBy } from 'lodash-es';
+import React, { useMemo, useState, useCallback } from 'react';
 import { generatePath, useNavigate, useLocation } from 'react-router-dom';
 
 import TextInput from '@/components/elements/input/TextInput';
 import { clientName } from '@/modules/hmis/hmisUtil';
 import { Routes } from '@/routes/routes';
-import { useOmniSearchClientsQuery } from '@/types/gqlTypes';
+import {
+  useOmniSearchClientsQuery,
+  useOmniSearchProjectsQuery,
+} from '@/types/gqlTypes';
 
 const OmniSearch: React.FC = () => {
   const [value, setValue] = useState<string | null>('');
   const navigate = useNavigate();
   const location = useLocation();
 
-  const { data, loading } = useOmniSearchClientsQuery({
-    variables: { input: { textSearch: value } },
-    skip: !value,
-  });
+  const { data: clientsData, loading: clientsLoading } =
+    useOmniSearchClientsQuery({
+      variables: { input: { textSearch: value } },
+      skip: !value,
+    });
+  const { data: projectsData, loading: projectsLoading } =
+    useOmniSearchProjectsQuery({
+      variables: { input: { textSearch: value } },
+      skip: !value,
+    });
+
+  const options = useMemo(() => {
+    const clients = clientsData?.clientSearch?.nodes || [];
+    const projects = projectsData?.projectSearch?.nodes || [];
+    return sortBy([...clients, ...projects], '__typename');
+  }, [clientsData, projectsData]);
+
+  const loading = clientsLoading || projectsLoading;
 
-  const clients = data?.clientSearch?.nodes || [];
+  const getOptionTargetPath = useCallback(
+    (option: NonNullable<typeof options>[number]) => {
+      let targetPath: string | null = null;
+      if (option.__typename === 'Client') {
+        targetPath = generatePath(Routes.CLIENT_DASHBOARD, {
+          clientId: option.id,
+        });
+      }
+      if (option.__typename === 'Project') {
+        targetPath = generatePath(Routes.EDIT_PROJECT, {
+          projectId: option.id,
+        });
+      }
+      return targetPath;
+    },
+    []
```

Comment:
As a rule I try to move functions that don't have dependencies outside of the component, so they're not redeclared every render. (like this callback and the one below). The performance impact is probably not noticable, though, so it's more of a style thing. Fine with leaving as-is

Reply:
@gigxz I remember why I did this now. I wanted to use `typeof options` in the type definitions so I could take advantage of the TS type detection on the `options` variable. Since that's defined inside the component body, I had to put these here to make that work. If I move them outside the component I'd have to statically type them, which is fine, but I like that a lot less than being able to use the inferred type from `options`.

Reply:
Also there should be pretty much zero performance impact to have them in the component with no deps. That should actually prevent them from being redeclared every render, since `useCallback` will only redeclare them if the deps change. No deps means it'll declare them on component mount and then keep that reference through the entire lifetime of the component.

Reply:
Ok sounds good! yeah it will keep the same reference to the function, but it will still redeclare the variable--  agreed near zero impact, more of a style thing like I said. keeping it here for the types makes sense. lgtm to merge once the backend PR goes in!